phoenix-core/src/it/java/org/apache/phoenix/end2end/ServerMetadataCacheTestImpl.java

```diff
@@ -0,0 +1,66 @@
+package org.apache.phoenix.end2end;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.hbase.ServerName;
+import org.apache.phoenix.cache.ServerMetadataCacheImpl;
+import org.apache.phoenix.hbase.index.util.ImmutableBytesPtr;
+import org.apache.phoenix.util.QueryUtil;
+import org.apache.phoenix.util.SchemaUtil;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+
+/**
+ * Implementation of {@link ServerMetadataCache} for Integration Tests.
+ * Supports keeping more than one instance keyed on the regionserver ServerName.
+ */
+public class ServerMetadataCacheTestImpl extends ServerMetadataCacheImpl {
```

Comment:
Can you add some more context in the comment something like:

PhoenixRegionServerEndpoint is a region server coproc. There is a 1-1 correspondence between PhoenixRegionServerEndpoint and ServerMetadataCache.
In ITs we can have multiple regionservers per cluster so we need multiple instances of ServerMetadataCache.
And HighAvailabilityTestingUtility creates 2 clusters so we need to have one instance of ServerMetadataCache for each regionserver in each cluster.

Reply:
done.